src/redux/reducers/compare.ts

```diff
@@ -0,0 +1,385 @@
+import {PayloadAction, TaskAbortError, createSlice, isAnyOf} from '@reduxjs/toolkit';
+
+import log from 'loglevel';
+
+import {K8sResource} from '@models/k8sresource';
+
+import {startAppListening} from '@redux/listeners/base';
+import {compareResources} from '@redux/services/compare/compareResources';
+import {fetchResources} from '@redux/services/compare/fetchResources';
+
+/* * * * * * * * * * * * * *
+ * State definition
+ * * * * * * * * * * * * * */
+export type CompareState = {
+  isOpen: boolean;
+  views: SavedDiffView[];
+  current: {
+    view: DiffView | SavedDiffView;
+    left?: ResourceSetData;
+    right?: ResourceSetData;
+    diff?: DiffData;
+    selection: string[];
+    viewDiff?: string; // comparisonId
+  };
+};
+
+export type CompareSide = 'left' | 'right';
+
+export type DiffData = {
+  loading: boolean;
+  comparisons: ResourceComparison[];
+};
+
+export type ResourceSetData = {
+  loading: boolean;
+  error: boolean;
+  resources: K8sResource[];
+};
+
+export type CompareOperation = 'union' | 'intersection' | 'symmetricDifference' | 'leftDifference' | 'rightDifference';
+
+export type CompareFilter = {
+  namespace?: string;
+};
+
+export type DiffView = {
+  leftSet: ResourceSet | undefined;
+  rightSet: ResourceSet | undefined;
```

Comment:
Is there a better word for `set`? I think it's to confusing with the verb `to set`.